ide/csl.api/test/unit/src/org/netbeans/modules/csl/api/test/CslTestBase.java

```diff
@@ -2145,6 +2145,14 @@ protected void checkFolds(String relFilePath) throws Exception {
 
         ParserManager.parse(Collections.singleton(testSource), new UserTask() {
             public @Override void run(ResultIterator resultIterator) throws Exception {
+
+                // FoldingScanner#folds calls source.getDocument(false) and receive non null values on JDK 8.
+                // On JDK 11+ however the document is null which leads to test failures in FoldingTest#testPHPTags.
+                // This is likely a race condition which is more likely to occur on JDK 11+ since the test can be
+                // brute forced to passing on linux if restarted often enough. 
+                // This fixes it by making sure the document is open before folds are computed
+                assertNotNull(resultIterator.getSnapshot().getSource().getDocument(true));
+                
```

Comment:
since this is in CslTestBase I am wondering if this also stabilizes other tests too as side effect